### PR TITLE
Feature/captain mode

### PR DESCRIPTION
**Pull request by BenBrostoff**
~~Putting this PR out so we're both aware of branch status.~~ This is good to go I think 👍 .
@sharkiteuthis - take a look and tell me what you think. The big change here was to build in `solver_id` based locking and banning. I personally use this a lot in showdown (mainly wanting to ban a top player from captain but keep them at flex).

I did not add any exposure controls for this feature but figure we can do that later.

### CODE REVIEW

File: draftfast/optimize.py [hunk 1]
```diff
@@ -22,12 +22,27 @@ def run(rule_set: RuleSet,
         player_pool,
         player_settings,
     )
+
+    if rule_set.game_type == 'showdown':
+        if not optimizer_settings.showdown_teams or \
+                len(optimizer_settings.showdown_teams) != 2:
+            raise Exception(
```
Comment (BenBrostoff) [inline draftfast/optimize.py]: Why is this necessary? Couldn't the optimizer just pick this up from the player pool?

File: draftfast/optimizer.py [hunk 2]
```diff
@@ -107,7 +113,7 @@ def _set_player_constraints(self):
             if lb > ub:
                 raise InvalidBoundsException
 
-            if p.multi_position:
+            if p.multi_position or self.showdown:
```
Comment (BenBrostoff) [inline draftfast/optimizer.py]: @sharkiteuthis I *think* this is the same as what was happening with no_reuse_captain and set_single_captain (I believe that one is already handled by the position constraints that look to the rule set).

**Comment by BenBrostoff** [comment]
@sharkiteuthis Did a release that has this code but I'll hold off a few days just so you can take a look if you have time before merging to master.

**Comment by BenBrostoff** [comment]
@sharkiteuthis Going to merge this - definitely push back with PRs / issues if any of it seems unreasonable.

**Comment by sharkiteuthis** [comment]
This all looks good. I'm not happy with multi-position players in general, but that makes sense -- all players are effectively mutli-position in a showdown slate. I have a branch checked out to work on that.
